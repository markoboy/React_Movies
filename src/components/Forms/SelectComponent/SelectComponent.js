/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import MultiSelect from 'react-multi-select-component';
import './SelectComponent.scss';

export default function SelectComponent({
  options,
  value,
  onChange,
  placeholder,
  label,
}) {
  const overrideStrings = {
    selectSomeItems: placeholder,
  };

  return (
    <div className="form-element-wrapper">
      {label && <label className="form__label">{label}</label>}
      <MultiSelect
        overrideStrings={overrideStrings}
        options={options}
        value={value}
        onChange={onChange}
      />
    </div>
  );
}
