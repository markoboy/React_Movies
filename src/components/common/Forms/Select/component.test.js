import React from 'react';
import { render } from '@testing-library/react';

import SelectComponent from './component';

jest.mock('react-multi-select-component', () => () => 'Mock Component');
const mockFn = jest.fn();

describe('SelectComponent Component', () => {
  it('renders SelectComponent Component with provided label', () => {
    const { container } = render(
      <SelectComponent options={[]} onChange={mockFn} label="Select an element" />
    );

    expect(container).toMatchSnapshot();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });
});
