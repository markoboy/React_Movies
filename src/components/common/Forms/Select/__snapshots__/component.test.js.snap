// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectComponent Component renders SelectComponent Component with provided label 1`] = `
<div>
  <div
    class="form-element-wrapper"
  >
    <label
      class="form__label"
      for=""
    >
      Select an element
    </label>
    Mock Component
  </div>
</div>
`;
